### ISSUE TITLE
Review calendar design

### ISSUE THREAD

**Issue by noahmanger** [18F/fec-cms#222]
At some point, we should take a step back and review how the calendar is working including all of the feedback we've received from the public:

https://github.com/18F/FEC/issues/256
https://github.com/18F/FEC/issues/253
https://github.com/18F/FEC/issues/255

As well as these notes from Paul (which we might end up dealing with in the next update to datatable styles):
https://github.com/18F/fec-cms/issues/213
https://github.com/18F/fec-cms/issues/211

Keeping this in the backlog for now, but didn't want to lose sight of it.

**Comment by jenniferthibault** [comment]
I'm going to bring all discussion of these issues into this issue, rather than chasing multiple issue streams. Issues relating to the calendar, ranked by my perceived priority.

- https://github.com/18F/fec-style/issues/251 : So that colorblind users can tell events apart, distinguish event category by more than color
- https://github.com/18F/FEC/issues/256 : List view look and feel: add month list view, (similar to your quarter view, but just of events and don't lose the colors!)
- https://github.com/18F/fec-cms/issues/213 : Decrease size of banner
- https://github.com/18F/FEC/issues/248 and https://github.com/18F/FEC/issues/251 : Clarify behavior of today button
- https://github.com/18F/FEC/issues/260 : Color too saturated

I've been working on a mockup that touches on all of these. The groupings and labeling of each accordion is being worked on here: https://github.com/18F/fec-cms/pull/233 so let's continue to keep that separate. What I could tell from that discussion was that either way, we were going to have fewer than the original six categories. If we assign color/shape values to the parent categories, that allows us to simplify the visual system too.

**Calendar grid view, with fewer categories grouped.**
- This reduces the spacing of the banner, a pattern that would be carried over to the data tables
- Reduces color palette by reverting to the dark blue headers. Since the calendar is powered by the API, I like this subtle connection, even if we're the only ones who see it.
- This mockup shows a few ways we could present the events in a way that meets 508 standards (not relying just on color to distinguish events): I believe the approach shown on the 19th (shape and text with full color) or the 24th (shape and text monochromatic) are the strongest options. Definitely open to other approaches to tie events into how they show up besides small icons. All ears if you have ideas!


![calendar_monthly copy](https://cloud.githubusercontent.com/assets/11636908/13784028/10169f0a-eaa4-11e5-8312-c239d5f08e77.png)


**Calendar list view, full color, show by event**
- the "current month" button would bring the user back to the start of the current month. If you're currently viewing the current month, the button would be disabled. This give users a shortcut back to the current period in time when they use the arrow buttons to navigate to future or past events
- This shows events grouped by event type (and corresponding icon with full color)


![calendar_monthly copy 2](https://cloud.githubusercontent.com/assets/11636908/13784029/11a125a2-eaa4-11e5-9236-5e13fdd60d6f.png)


**Calendar list view, full color, show by date**
- This shows events grouped by date (and corresponding icon with full color)


![calendar_monthly copy 3](https://cloud.githubusercontent.com/assets/11636908/13784481/cfbd8f5c-eaa5-11e5-8416-b1673d6254ec.png)


**Calendar list view, monochromatic, show by date**
- This shows events grouped by date (and corresponding monochromatic icon)


![calendar_monthly copy 4](https://cloud.githubusercontent.com/assets/11636908/13784484/d0fd4484-eaa5-11e5-8032-1f38239d713c.png)

All thoughts welcome! If you'd like to put comments directly on any of the mockups, I've set up this InVision (link coming, my internet is crawling at the moment).

**Comment by noahmanger** [comment]
These are great. I really like the symbol approach. I think I like the full color style more than monochromatic, and I like the outlined all-day event style too.

Only bit of feedback I have is that even though elections and reporting periods are grouped, it feels a little weird to have them with the same icon just because they're such different types of events.

**Comment by emileighoutlaw** [comment]
I suspect Amy will feel the same about elections and reporting periods as Noah mentioned above. 

I like the full color style, too. My note (which is tiny) — rather than moving to `Current month`, how would you feel about keeping the button copy as `Today`? 

I enjoy that `Today` interaction so much (and I like that it's aligned with Google's calendar), but I'm open to hearing reasons why "current month" is better, if you have strong opinions. :)

**Comment by jenniferthibault** [comment]
After speaking with my critique group about this yesterday, (one of whom had made a calendar in govt and could identify with these challenges) they encouraged me to think about dropping the color/icon/event type visual signifiers altogether for a few reasons:
- if you're giving people the ability to filter by event type, the icon/color isn't as needed
- when you're working with such a small shape and space, icons can look similar to those with poor vision (think square>>circle, try a squint test)
- if we do want a signifier for every category, instead of fewer signifiers for the overall categories as I've had above, you lose the ability for users to quickly remember what category goes with what signifier because there are so many
- they posed a good question: do we really NEED this feature? Or is it complicating an interaction (filter sort) that already meets the need to find events in certain categories?

I think all of these points are solid, so I explored what that could look like on a full calendar of busy events:

![calendar_monthly copy 7](https://cloud.githubusercontent.com/assets/11636908/13816397/a2ade89e-eb64-11e5-8e1b-a1da8bacd1cb.png)

and then explored how we could tie in the category to the event details panel, which would be useful if a user hadn't applied filters yet, or had applied multiple filter categories:

![calendar_monthly copy 8](https://cloud.githubusercontent.com/assets/11636908/13816444/cb0babe6-eb64-11e5-92c5-4a4ee126005d.png)

Overall I think this approach is a lot easier on the eyes, especially compared to a full calendar of colored icons:
![calendar_monthly copy 6](https://cloud.githubusercontent.com/assets/11636908/13816696/ccd76f72-eb65-11e5-8b6e-0794ff539f95.png)


Would we be game to back off the color/visual signifier for events?

**Comment by emileighoutlaw** [comment]
I'm game (not that you _need_ my buy in)! This is an interesting thought!

**Comment by jenniferthibault** [comment]
I don't want to lose your thought on the button copy @emileighoutlaw – I'm down to keep it as "Today" if that doesn't prove confusing to users in testing. Since we dropped daily and weekly views, I was aiming for accuracy of timeframe that you'd be viewing.

**Comment by emileighoutlaw** [comment]
Yeah — I did take a look at Google, and they keep it as "Today" even on month view. But I see what you're saying. I'm hoping, since we do nice shading on the "today" of month view, that it'll make sense. 

This is a good interaction to keep in mind for usability testing (cc @brethauer).

**Comment by PaulClark2** [comment]
One of the FEC comments at the "today" button was internal users assumed that clicking that button would take you to a list of today's events.  However, the "today" button just takes you to the current month and highlights the current day.   @amykort @wluoFEC

**Comment by jenniferthibault** [comment]
@noahmanger hearing no objections to the design, the first designs here https://github.com/18F/fec-cms/issues/222#issuecomment-197367852 (sans event colors or symbols) can be turned over to front end implementation. Transferring the issue over to you. 

This adds a change to the event detail tooltip, where the category would be visible in the detail panel, so let me know if you want to pair up of talk through anything when you get there.

**Comment by noahmanger** [comment]
Resolved by https://github.com/18F/fec-cms/pull/256
